src/components/data-exploration/heatmap/HeatmapPlot.jsx

```diff
@@ -86,7 +88,18 @@ const HeatmapPlot = (props) => {
   }, [heatmapSettings]);
 
   useEffect(() => {
-    if (hierarchy.length === 0 || cellSetsLoading) {
+    const selectedGenesLoading = _.intersection(selectedGenes, loadingGenes).length > 0;
+
+    if (markerGenesLoading || selectedGenesLoading) {
```

Comment:
`markerGenesLoading` only happens the first time you visit data exploration
`selectedGenesLoading` happens every time you change the selected genes on the gene list and add/remove them to the marker heatmap

Reply:
Comments added

Reply:
have you pushed? I don't see them

Reply:
Oops, yea sorry I stashed and didn't include it,